src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/services/mapper/CaseDataBuilderTest.java

```diff
@@ -0,0 +1,162 @@
+                buildMajorStatusGivenStatusAndDate(GapsEvent.HEARING_POSTPONED.getStatus(), TEST_DATE2)
+            ))
+            .hearing(getHearing())
+            .minorStatus(Collections.singletonList(
+                buildMinorStatusGivenIdAndDate("26", TEST_DATE2)
+            ))
+            .build();
+
+        events = caseDataBuilder.buildEvent(appealCaseWithMinorStatusId26AndPostponedWithSameDates);
+
+        assertEquals("Events size should be 2 here", 2, events.size());
+        int actualNumberOfPostponedEventsWithSameDate = events.stream()
+            .filter(event -> event.getValue().getType().equals(GapsEvent.HEARING_POSTPONED.getType()))
+            .filter(event -> event.getValue().getDate().equals(
+                ZonedDateTime.parse(TEST_DATE2).toLocalDateTime().toString()))
+            .collect(Collectors.toList())
+            .size();
+        assertEquals("Only one postponed event with same minor status date expected here", 1,
+            actualNumberOfPostponedEventsWithSameDate);
+    }
+
+    @Test
+    public void whenBuildEventMethodIsCalledThenItReturnsAnEventListSortedByDateInDescOrder() {
+        AppealCase appealCaseWithMinorStatusId26AndMajorStatuses = AppealCase.builder()
+            .appealCaseCaseCodeId("1")
+            .majorStatus(Collections.singletonList(
+                buildMajorStatusGivenStatusAndDate(GapsEvent.APPEAL_RECEIVED.getStatus(), TEST_DATE)
+            ))
+            .hearing(getHearing())
+            .minorStatus(Collections.singletonList(
+                buildMinorStatusGivenIdAndDate("26", TEST_DATE2)
+            ))
+            .build();
+
+        events = caseDataBuilder.buildEvent(appealCaseWithMinorStatusId26AndMajorStatuses);
+
+        assertTrue("events size only has 1 element", events.size() > 1);
+        Event actualMostRecentEvent = events.get(0).getValue();
+        assertEquals("expected most recent Event is wrong",
+            actualMostRecentEvent.getType(), GapsEvent.HEARING_POSTPONED.getType());
+    }
+
+    @Test
+    public void givenAFewMinorStatuesShouldCreatePostponedEventFromTheLatestMinorStatus() {
+        AppealCase appealCaseWithTwoMinorStatusId26WithDifferentDates = AppealCase.builder()
+            .appealCaseCaseCodeId("1")
+            .majorStatus(Collections.singletonList(
+                buildMajorStatusGivenStatusAndDate(GapsEvent.APPEAL_RECEIVED.getStatus(), TEST_DATE)
+            ))
+            .hearing(getHearing())
+            .minorStatus(Arrays.asList(
+                buildMinorStatusGivenIdAndDate("26", TEST_DATE),
+                buildMinorStatusGivenIdAndDate("26", TEST_DATE2)
+            ))
+            .build();
+
+        events = caseDataBuilder.buildEvent(appealCaseWithTwoMinorStatusId26WithDifferentDates);
+
+        assertEquals("latest event expected here is postponed",
+            events.get(0).getValue().getType(), GapsEvent.HEARING_POSTPONED.getType());
+    }
```

Comment:
It was decided yesterday that minor status code 26 need not be handled for postponed event. Please remove them.

Reply:
will be addressed here -> https://tools.hmcts.net/jira/browse/SSCS-3139